examples/js/loaders/STLLoader.js

```diff
@@ -254,53 +280,78 @@ THREE.STLLoader.prototype = {
 
 			var result;
 
-			while ( ( result = patternFace.exec( data ) ) !== null ) {
-
-				var vertexCountPerFace = 0;
-				var normalCountPerFace = 0;
+			var groupVertexes = [];
+			var groupCount = 0;
+			var startVertex = 0;
+			var endVertex = 0;
 
-				var text = result[ 0 ];
+			while ((result = patternSolid.exec(data)) !== null) {
 
-				while ( ( result = patternNormal.exec( text ) ) !== null ) {
+				startVertex = endVertex;
 
-					normal.x = parseFloat( result[ 1 ] );
-					normal.y = parseFloat( result[ 2 ] );
-					normal.z = parseFloat( result[ 3 ] );
-					normalCountPerFace ++;
+				solid = result[0];
```

Comment:
Your code currently throws a runtime errors since `solid` is undefined. Please test your code with the official example to avoid such errors.

Reply:
My apologies. I will fix that. I am not sure how I missed that: it did not throw in my app, but obviously it makes no sense.

Reply:
Corrected in commit 3ea613ad256b814b7e48bb5c5151dc80cd166ed7. 
The example in three runs:

![image](https://user-images.githubusercontent.com/423032/63227527-2e315a00-c1ad-11e9-9e2b-55c291ce6d3c.png)